Description: lxc-ubuntu: use relative path as target for bind mount
 An absolute path will be interpreted as absolute with respect to the
 parent's namespace.
 This patch will be forwarded.
Author: Serge Hallyn <serge.hallyn@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/lxc/+bug/968371
Forwarded: yes

Index: lxc-0.8.0~rc1/templates/lxc-ubuntu.in
===================================================================
--- lxc-0.8.0~rc1.orig/templates/lxc-ubuntu.in	2012-04-23 23:03:05.015947121 -0500
+++ lxc-0.8.0~rc1/templates/lxc-ubuntu.in	2012-04-23 23:04:13.823945956 -0500
@@ -543,7 +543,12 @@
     # bind-mount the user's path into the container's /home
     h=`getent passwd $user | cut -d: -f 6`
     mkdir -p $rootfs/$h
-    echo "$h $h none bind 0 0" >> $path/fstab
+	# use relative path in container
+	h2=${h#/}
+	while [ ${h2:0:1} = "/" ]; do
+		h2=${h2#/}
+	done
+    echo "$h $h2 none bind 0 0" >> $path/fstab
 
     # Make sure the group exists in container
     grp=`echo $pwd | cut -d: -f 4`  # group number for $user
